### ISSUE TITLE
Atom port dead?

### ISSUE THREAD

**Issue by Karl273** [msiniscalchi/atom-latextools#158]
Dear developers,

I notice that no commits have been made for a long time for an Atom port.
Is any development of the port planned nonetheless?
Any help needed?
Best,
Karl.

**Comment by ig0774** [comment]
It's not completely dead, but I haven't had a lot of time to work on it lately. Patches, etc. are welcome, though.

**Comment by Karl273** [comment]
I see. Btw, a port to VSC might become useful and popular. Currently, there is only latex-workshop there, which desperately lacks functionality, as compared to latextools.

**Comment by r-stein** [comment]
I think the main issue with the atom port is that the developers are still using Sublime Text for LaTeX, because it has a better syntax highlighting and more features. Therefore on the one side the motivation to add features to Atom isn't too hight and on the other side the motivation to switch to Atom is low.
In VSCode imo it would be better to improve the existing package instead of spawning more packages, which needs to reimplement a lot of functionalities.

**Comment by werunom** [comment]
Sad to hear that there package is not being actively developed anymore. This was a really good comprehensive package for latex. 
Would be glad to know if there are any plans for further development.

**Comment by cortner** [comment]
maybe somebody wants to fork it? I would have made 2-3 PRs already, but seeing that none of the existing ones are being accepted and my questions in the Issues left unanswered, I didn't bother. But if anybody would maintain a fork, say `latextools2`, then I'd try to help out every ones in a while, and I the PRs show many others would as well.

**Comment by shanepelletier** [comment]
I've started the process of forking over at https://github.com/latextools/latextools-atom, I haven't had time to actually get the code over there yet though. If any of the collaborators of this repo want to join the effort over there I can add you to the org.

**Comment by r-stein** [comment]
If someone is interested in supporting LaTeXTools it would be a better idea to add him as collaborator than forking the repo and starting an other competing package, which may have less support in the future even if it has more support for the moment (as LaTeXing in Sublime Text). This wasn't a real problem for Sublime Text and I doubt it will one for Atom.

@shanepelletier 
If don't want to go this way, but create your own package you should (and maybe must) publish it with a different name and it would be nice to rename the github org correspondingly.

**Comment by cortner** [comment]
I've only used `latextools` since I switched to Atom, but does anybody here have experience with the other latex packages for Atom? How do they compare re functionality?

**Comment by shanepelletier** [comment]
Tl;dr: I'm mostly waiting to see what happens with this report. 

@r-stein I'd be fine with maintaining this repo but seeing as the current maintainers seem to be mostly ignoring it I decided to fork rather than wait around. I'll see if it's possible to remove the Atom package once it's published and if so I'll publish under another name, otherwise I'll hold off until I get in contact with the current maintainers. I don't want to publish a semi-competing package and pollute the namespace if I can help it.

**Comment by r-stein** [comment]
I am one of the (3) Sublime Text maintainers so I answered with the insights I have and by writing in this issue you have contact to the maintainers.
However creating a latextools org and forking (and publishing?) without even adding a comment offering your participation before doesn't really help to build the confidence, which is needed for a collaborator (behave responsible with the repo, discuss major changes before doing them, ...).

**Comment by cortner** [comment]
Fwiw I suggested making 2-3 minor changes (5-10 lines) in a few issues but never even got a response. I appreciate people can get busy but just abandoning a project that is used by so many people should mean it can be forked and
Continued by somebody else.

**Comment by r-stein** [comment]
@cortner 
I fully understand your frustration and think it is a good idea to fork and continue dead packages. However the usual way would be to contact the author(s) about it (as you did) and (as I suggested) there may even be a better alternative. If they don't answer for at least 14 days you may try to undertake the package. If you don't come to an agreement you may create your own package based on the source code (MIT license).

Creating an org based on the package name, forking (and publishing?) the package so the maintainers will contact you isn't really a good base for an agreement. To emphasize this: The creation of an github org based on a actively-developed Sublime Text package and it's (unfortunately) not actively-developed Atom port bothers me the most.

**Comment by shanepelletier** [comment]
I screwed up.

I stumbled on this repo after about five minutes of searching for a good way to edit LaTeX in Atom, found that none of the maintainers had been active on GitHub for a couple months, and that someone was wondering if it was dead. I decided that the best way to save it would be to fork it and manually merge at least some of the pull requests on this repo. I created an org for latextools since I expected that other people would want to collaborate on the revival of the package and possibly even take over the duty of maintaining the package. latextools made sense to me as the name of the org, since I envisioned eventually having a latextools port for many different text editors and felt that they should all be under the same branding.

I hadn't done much research on the original Sublime Text version beyond taking a glance at the commit history on GitHub, which made the original package look just as dead as this one. I should've taken a glance at the issues to double check. I planned on working with the maintainers of the Sublime Text package to see if they'd like to move their code over to the latextools org, or to eventually fork the repo myself if none of the maintainers responded after a reasonable amount of time (around a month).

I've changed the name of the org to easytex and the name of the package to easytex-atom, which I feel better reflects its purpose anyway: to make editing LaTeX easier in whatever editor someone chooses to use. I'm still not sure I'll be publishing the package anytime soon, I'd like to get a few more pull requests merged in and possibly edit some of the documentation before I do so.

**Comment by cortner** [comment]
@r-stein fair point about the org and package-name in general. But also can we please give @shanepelletier a break. Whether his judgement was perfect or not, I think it is fair to say he acted in good faith. And indeed just cleared this up. 

@shanepelletier : I hope you can make this work as I quite like latextools

**Comment by r-stein** [comment]
Even if it wasn't the best start doesn't mean it won't have a good end :) Since @shanepelletier has explained his motivation, renamed his org and project, and is understanding, I am okay with it and won't force a conflict. We may still come to an agreement and satisfying communication. Please don't publish the package until then.

Aside from that we are preparing a new major version on ST, so we are working on a different branch: https://github.com/SublimeText/LaTeXTools/commits/version4

**Comment by shanepelletier** [comment]
I'm planning on opening an issue on the original LaTeXTools package to discuss either moving that package (which I fully understand if you and the other maintainers are against) or renaming/refocusing the easytex org, and figuring out what I'm (possibly we're) going to do about the Atom port. I think the discussion should happen both on this repo and over there since the ST version is the original and did come up with the name. I'll hold off on publishing until things are cleared up between all of us.

**Comment by r-stein** [comment]
@shanepelletier 
In Sublime Text the LaTeXTools package is (for historical reasons) part of the Sublime Text org, which consists of over 50 Sublime Text package developers. Therefore it will have a minimal support even if every collaborator stops being active. So moving this to an other org would be a huge step.
I think the core of this discussion is that you are willing/want to maintain this (or a different LaTeX package for Atom) and we want/need and an additional maintainer. Since we are all in favor of adding you as a collaborator I will write you an email to discuss the details.

**Comment by shanepelletier** [comment]
@r-stein that's fair, just for visibility it's probably better off as part of that org anyways.

**Comment by cortner** [comment]
is anything happening about continuing this `atom-latextools`?

**Comment by shanepelletier** [comment]
The maintainers wanted me to give the Sublime Text version a shot before they would add me as a collaborator. Unfortunately, I ended up dropping the project I was working on that required LaTeX so I didn't have a chance to give it a shot, thus preventing me from being added as a collaborator.

**Comment by cortner** [comment]
Too bad
